Sources/AWSSDKSwiftCore/MetaDataService.swift

```diff
@@ -89,7 +89,7 @@ struct MetaDataService {
     }
 
     static func request(host: String, uri: String, timeout: TimeInterval) -> Future<HTTPClient.Response> {
-        let client = HTTPClient(hostname: host, port: 80)
```

Comment:
If we are relying on the uri for the host / port info, then I think MetaDataServiceProvider#uri needs to return the baseURLString for .ecsCredentials instead of the containerCredentialsUri which does not include the host / port.

Reply:
The HTTPClient will default to port 80 if an https:// scheme prefix isn't found so this should work ok. Haven't setup an ECS container before so not sure what we get back from "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI" environment variable. Does it not include the host info?

Reply:
correct `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` is just the path.

Reply:
Does the change fix this? 

The MetaDataServiceProvider enum is a bit confusing. baseURLString, uri mean different things for different providers. Would be better as a protocol and individual classes
```
protocol MetaDataServiceProvider {
    public static func getCredential() throws -> Future<CredentialProvider>
}

extension MetaDataServiceProvider {
    static func request(host: String, uri: String, timeout: TimeInterval) -> Future<HTTPClient.Response> {
        ...
    }
}

class ECSMetaDataService: MetaDataServiceProvider {
    public static func getCredential() throws -> Future<CredentialProvider> {
    }
}

class InstanceProfileMetaDataService: MetaDataServiceProvider {
    public static func getCredential() throws -> Future<CredentialProvider> {
    }
}

```

Reply:
@adam-fowler yes, the change fixes it.

Agreed it's confusing. I like your suggestions. Feel free, and I'll put it on my todo as well.

Reply:
@jonnymacs You good for this one to go in?